lib/proxy.js

```diff
@@ -137,12 +137,13 @@ function buildResultingMiddleware (host, options, stream) {
     var reqOpts = {
       url     :  URL.format(proxyUrl),
       method  : req.method,
-      headers : _.defaults(options.headers || {}, req.headers),
+      // headers : _.defaults(options.headers || {}, req.headers),
+      headers : options.headers,
```

Comment:
it wouldnt he hard to write our own. just iterate and make undefined delete it?

I'm actually thinking defaults does that. it says it overrides for every 'enumerable property' and setting a property to undefined still shows up the array of keys for the given object. I'd test it myself but I'm not near my ubuntu machine at the moment.

if it helps clarify what I mean, here my key test from the console 
```javascript
Object.keys({test : undefined})
["test"]
```

https://lodash.com/docs#defaults